k256/src/arithmetic/field.rs

```diff
@@ -516,6 +517,34 @@ mod tests {
         }
     }
 
+    #[test]
+    #[ignore] // TODO(tarcieri): debug failure
+    fn two_inv_constant() {
+        assert_eq!(
+            FieldElement::from(2u64) * FieldElement::TWO_INV,
+            FieldElement::ONE
+        );
+    }
+
+    #[test]
+    #[ignore] // TODO(tarcieri): debug failure
+    fn root_of_unity_constant() {
+        // ROOT_OF_UNITY^{2^s} mod m == 1
+        assert_eq!(
+            FieldElement::ROOT_OF_UNITY.pow_vartime(&[1u64 << FieldElement::S, 0, 0, 0]),
+            FieldElement::ONE
+        );
+    }
+
+    #[test]
+    #[ignore] // TODO(tarcieri): debug failure
+    fn root_of_unity_inv_constant() {
+        assert_eq!(
+            FieldElement::ROOT_OF_UNITY * FieldElement::ROOT_OF_UNITY_INV,
+            FieldElement::ONE
+        );
+    }
```

Comment:
@fjarri guess you were right about a cross-check because all of these are failing.

I'll try recalculating them.

Reply:
Huh, I was pretty sure I did it correctly. That's weird.

Reply:
Are you sure there's no normalization missing? I looked in the code and it seems `FieldElement` does not automatically normalize.

Reply:
I tried adding normalization in [84adaa4](https://github.com/RustCrypto/elliptic-curves/pull/737/commits/84adaa4e2d7869a309a7860e239c8a4f65fd8605). It fixed the tests for the constant inversions, but the root of unity constant test is still failing (possibly a normalization-related issue related to `Field::pow_vartime`?)

Reply:
Went through all of the constants and found the problem: `S` was `0` where it should be 1. `m - 1` is the following in binary:

```
1111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111011111111111111111111110000101110
```

`S` is the number of trailing zeroes (which is somewhat confusingly described as "leading zero bits in the little-endian bit representation" in the `ff` docs /cc @str4d

So in this case, it's 1. With that changed, the `k256` tests pass.

Reply:
It seems that all the values I calculated were correct then, that's a relief.